core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -313,35 +313,76 @@ class RequestSendThread(val controllerId: Int,
   }
 }
 
-class ControllerBrokerRequestBatch(config: KafkaConfig,
-                                   controllerChannelManager: ControllerChannelManager,
-                                   controllerEventManager: ControllerEventManager,
-                                   controllerContext: ControllerContext,
-                                   stateChangeLogger: StateChangeLogger)
-  extends AbstractControllerBrokerRequestBatch(config, controllerContext, stateChangeLogger) {
+class ControllerBrokerRequestBatch(
+  config: KafkaConfig,
+  controllerChannelManager: ControllerChannelManager,
+  controllerEventManager: ControllerEventManager,
+  controllerContext: ControllerContext,
+  stateChangeLogger: StateChangeLogger
+) extends AbstractControllerBrokerRequestBatch(
+  config,
+  () => controllerContext,
+  () => config.interBrokerProtocolVersion,
+  stateChangeLogger
+) {
 
   def sendEvent(event: ControllerEvent): Unit = {
     controllerEventManager.put(event)
   }
-
   def sendRequest(brokerId: Int,
                   request: AbstractControlRequest.Builder[_ <: AbstractControlRequest],
                   callback: AbstractResponse => Unit = null): Unit = {
     controllerChannelManager.sendRequest(brokerId, request, callback)
   }
 
+  override def handleLeaderAndIsrResponse(response: LeaderAndIsrResponse, broker: Int): Unit = {
+    sendEvent(LeaderAndIsrResponseReceived(response, broker))
+  }
+
+  override def handleUpdateMetadataResponse(response: UpdateMetadataResponse, broker: Int): Unit = {
+    sendEvent(UpdateMetadataResponseReceived(response, broker))
+  }
+
+  override def handleStopReplicaResponse(stopReplicaResponse: StopReplicaResponse, brokerId: Int,
+                                         partitionErrorsForDeletingTopics: Map[TopicPartition, Errors]): Unit = {
+    if (partitionErrorsForDeletingTopics.nonEmpty)
+      sendEvent(TopicDeletionStopReplicaResponseReceived(brokerId, stopReplicaResponse.error, partitionErrorsForDeletingTopics))
+  }
+}
+
+abstract class ControllerBrokerRequestMetadata {
```

Comment:
Few things with this one ^

* Can it be a trait? 
* Can we avoid "Metadata" in the name? Maybe ControllerBrokerRequestContext?
* Can we define it in a separate file?